File: camerax_platform_interface/lib/src/camera_selector.dart
import 'lens_facing.dart';

final class CameraSelector {
  final LensFacing lensFacing;

  const CameraSelector({
    required this.lensFacing,
  });

  static const CameraSelector front = CameraSelector(
    lensFacing: LensFacing.front,
  );
  static const CameraSelector back = CameraSelector(
    lensFacing: LensFacing.back,
  );
  static const CameraSelector external = CameraSelector(
    lensFacing: LensFacing.external,
  );

  @override
  bool operator ==(Object other) {
    return other is CameraSelector && other.lensFacing == lensFacing;
  }

  @override
  int get hashCode => lensFacing.hashCode;
}

File: camerax_ios/ios/Classes/CameraSelector.swift
//
//  CameraSelector.swift
//  camerax_ios
//
//  Created by 闫守旺 on 2024/7/12.
//

import Foundation

@objc public class CameraSelector: NSObject {
    @objc public let lensFacing: LensFacing
    
    @objc public init(lensFacing: LensFacing) {
        self.lensFacing = lensFacing
    }
    
    @objc public static let front = CameraSelector(lensFacing: .front)
    @objc public static let back = CameraSelector(lensFacing: .back)
    @objc public static let external = CameraSelector(lensFacing: .external)
}
